src/torcontrol.cpp

```diff
@@ -524,11 +524,15 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
 
         // Now that we know Tor is running setup the proxy for onion addresses
         // if -onion isn't set to something else.
-        if (gArgs.GetArg("-onion", "") == "") {
+        // Or if we have already an outbound proxy defined. we skip this footgun gracefully.
+        proxyType addrOnion;
+        if (gArgs.GetArg("-onion", "") == "" && !GetProxy(NET_ONION, addrOnion)) {
             CService resolved(LookupNumeric("127.0.0.1", 9050));
-            proxyType addrOnion = proxyType(resolved, true);
-            SetProxy(NET_ONION, addrOnion);
-            SetReachable(NET_ONION, true);
+            addrOnion = proxyType(resolved, gArgs.GetBoolArg("-proxyrandomize", true));
+            if (SetProxy(NET_ONION, addrOnion)) {
+                LogPrint(BCLog::TOR, "tor: Default outbound tor .onion address proxy is set to 127.0.0.1:9050\n");
+                SetReachable(NET_ONION, true);
+            } else LogPrint(BCLog::TOR, "tor: Error default outbound tor .onion address proxy for 127.0.0.1:9050 failed to create\n");
```

Comment:
```suggestion
            } else {
                LogPrint(BCLog::TOR, "tor: Failed to create default outbound tor .onion address proxy for 127.0.0.1:9050\n");
            }
```

Reply:
thx, had run it trough clang-format-diff and it also re-formated the net enum,  will come up with test and clang-diff fixup push that we that can autosquash ... , if you want to be co, i could commit the suggest as is, since i see it the same way, my while dev sarcasm comments must not be merged in the final version?

Reply:
Also i think about, split the tor.md from this PR since wording yields always to intensive discussion.